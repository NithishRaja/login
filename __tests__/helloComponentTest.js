import {shallow} from "enzyme";
import React from "react";
import ReactDOM from "react-dom";
import "react-test-renderer";
import Hello from "./../src/js/components/helloComponent";

describe("<Hello />", () => {

  it("should render a message and a button", () => {
    const wrapper = shallow(
      <Hello />
    );
    expect(wrapper).toMatchSnapshot();
  });

});
