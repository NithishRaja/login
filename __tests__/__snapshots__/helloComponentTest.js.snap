// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> should render a message and a button 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section>
    <h1>
        Hello 
        !!
    </h1>
    <button
        id="toggleMessage"
    >
        change message
    </button>
</section>,
  "nodes": Array [
    <section>
      <h1>
            Hello 
            !!
      </h1>
      <button
            id="toggleMessage"
      >
            change message
      </button>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Hello />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Hello {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section>
          <h1>
                    Hello 
                    !!
          </h1>
          <button
                    id="toggleMessage"
          >
                    change message
          </button>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section>
          <h1>
                    Hello 
                    !!
          </h1>
          <button
                    id="toggleMessage"
          >
                    change message
          </button>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Hello />,
}
`;
